static/js/components/forms/B2BPurchaseForm.js

```diff
@@ -35,89 +42,152 @@ export const validate = (values: Object) => {
   return errors
 }
 
-const B2BPurchaseForm = ({ onSubmit, products, requestPending }: Props) => (
-  <Formik
-    onSubmit={onSubmit}
-    initialValues={{
-      num_seats: "",
-      email:     "",
-      product:   ""
-    }}
-    validate={validate}
-    render={({ values }) => {
-      let itemPrice, totalPrice, numSeats
-      const productId = parseInt(values.product)
-      const product = products.find(product => product.id === productId)
-      const productVersion = product ? product.latest_version : null
-      if (productVersion) {
-        numSeats = parseInt(values.num_seats)
-        if (!isNaN(numSeats) && productVersion.price !== null) {
-          itemPrice = new Decimal(productVersion.price)
-          totalPrice = itemPrice * new Decimal(numSeats)
+class B2BPurchaseForm extends React.Component<Props> {
+  applyCoupon = async (
+    event: Event,
+    values: Object,
+    setFieldError: Function
+  ) => {
+    const { clearCouponStatus, fetchCouponStatus } = this.props
+
+    event.preventDefault()
+
+    if (!values.coupon) {
+      clearCouponStatus()
+      return
+    }
+
+    if (!values.product) {
+      setFieldError("coupon", "No product selected")
+      return
+    }
+
+    const response = await fetchCouponStatus({
+      product_id: values.product,
+      code:       values.coupon
```

Comment:
Nit: I had a space leading my coupon code and it failed because of that. Can you trim the string in this form or in the view on the backend?